### PR TITLE
Fix: display allocated SAFE amount

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves
Displays SAFE token allocation properly.
Ported from https://github.com/safe-global/safe-react/pull/4081.

### CODE REVIEW

File: src/components/common/SafeTokenWidget/index.tsx [hunk 1]
```diff
@@ -44,10 +44,7 @@ const SafeTokenWidget = () => {
       }
     : undefined
 
-  const safeBalance = balances.balances.items.find((balanceItem) => balanceItem.tokenInfo.address === tokenAddress)
-
-  const safeBalanceDecimals = Number(safeFormatUnits(safeBalance?.balance || 0, safeBalance?.tokenInfo.decimals))
-  const flooredSafeBalance = formatAmountWithPrecision(safeBalanceDecimals, 2)
+  const flooredSafeBalance = formatVisualAmount(allocation, DECIMALS)
```
Comment (DiogoSoaress) [inline src/components/common/SafeTokenWidget/index.tsx:47]: Don't we want to keep the 2 decimals precision? It broke the tests

**Review by DiogoSoaress — CHANGES_REQUESTED** [review]

File: src/components/common/SafeTokenWidget/index.tsx (hunk 1 above)
Comment (katspaugh) [reply src/components/common/SafeTokenWidget/index.tsx:47]: It's the default, and it's not what broken them.

**Review by DiogoSoaress — APPROVED** [review]

File: cypress/e2e/smoke/address_book.cy.js [hunk 2]
```diff
@@ -30,8 +29,7 @@ describe('Address book', () => {
       // Add a new entry manually
       cy.contains('Create entry').click()
       cy.get('input[name="name"]').type(NAME)
-      cy.get('input[name="address"]').type(ENS_NAME)
-      // cy.wait(5000)
+      cy.get('input[name="address"]').type(`gor:${ENS_ADDRESS}`)
```
Comment (DiogoSoaress) [inline cypress/e2e/smoke/address_book.cy.js:32]: This seems to be a bug in the form. I was investigating on it but I will fill in an issue

**Comment by francovenica** [comment]
The widget is there and it shows the correct value

Some details (I can move these into a new ticket necessary)

- The widget is not clickable, maybe because the app is not available in this PR
- The number has more than 2 decimals. In prod is capped at 2 decimals.
- Is not visible for a mobile view anywhere. I think we should show it somewhere for a while until more people claim their tokens

File: src/components/common/SafeTokenWidget/index.tsx (hunk 1 above)
Comment (katspaugh) [reply src/components/common/SafeTokenWidget/index.tsx:47]: @DiogoSoaress you were actually right, it needs the precision, as Franco also pointed out. I've passed it explicitly.

**Comment by katspaugh** [comment]
@francovenica I've fixed the decimals.
The claiming app isn't tagged on staging, please use the prod toggle.
Regarding mobile, there's no place for it in the header. I will be solved in near future by a dedicated Dashboard widget.

File: src/components/common/SafeTokenWidget/__tests__/SafeTokenWidget.test.tsx [hunk 3]
```diff
@@ -81,50 +65,21 @@ describe('SafeTokenWidget', () => {
...
-      error: undefined,
-    }))
+  it('Should display 0 if Safe has no SAFE token', async () => {
+    ;(useSafeTokenAllocation as jest.Mock).mockImplementation(() => BigNumber.from(0))
 
     const result = render(<SafeTokenWidget />)
     await waitFor(() => expect(result.baseElement).toHaveTextContent('0'))
   })
 
   it('Should display the value formatted correctly', async () => {
```
Comment (DiogoSoaress) [inline src/components/common/SafeTokenWidget/__tests__/SafeTokenWidget.test.tsx:75]: This test should cover the precision nuance.
I.e, without explicitly passing the precision it renders with a wrong number of decimals **and** when explicitly passing the precision, it formats the number according to that

**Review by DiogoSoaress — CHANGES_REQUESTED** [review]

File: src/components/common/SafeTokenWidget/__tests__/SafeTokenWidget.test.tsx (hunk 3 above)
Comment (katspaugh) [reply src/components/common/SafeTokenWidget/__tests__/SafeTokenWidget.test.tsx:75]: I've set [a better mock value](https://github.com/safe-global/web-core/pull/908/commits/55a334fe0f2c1d615777b48873f10f06428a5c34#diff-cfe34aabfe1308977e7ecf7c7128682037147b447afa6badc9035bf46c10f983R76) for the allocation amount, that results in 4 decimals if formatted badly.

**Review by DiogoSoaress — APPROVED** [review]

**Pull request merged**